fuel/datasets/hdf5.py

```diff
@@ -64,3 +67,127 @@ def get_data(self, state=None, request=None):
                 raise ValueError
         data = [node[request] for node in self.nodes]
         return data
+
+
+@do_not_pickle_attributes('data_sources')
+class H5PYDataset(Dataset):
+    """An h5py-fueled HDF5 dataset.
+
+    This dataset class assumes a particular file layout:
+
+    * Data sources reside in the root group, and their names define the
+      source names.
+    * The dataset is not explicitly split. Instead, splits are defined as
```

Comment:
Got it, although this will only work here if all of the atomic sets can be
arranged such that any concatenated set you might like will be contiguous,
right?